pyvrp/crossover/tests/test_selective_route_exchange.py

```diff
@@ -127,74 +127,30 @@ def test_srex_sorts_routes():
             assert_equal(permuted_offspring, offspring)
 
 
-def test_srex_greedy_repair():
+def test_srex_does_not_reinsert_unplanned_clients():
     """
-    Tests the case where greedy repair is used during SREX crossover.
+    Tests the case where the routes are not perfectly complementary. In that
+    case, some clients will no longer in the solution after crossover: SREX
+    does not reinsert those, but instead leaves that to the search method.
     """
     data = read("data/OkSmallGreedyRepair.txt")
     cost_evaluator = CostEvaluator(20, 6)
 
+    # Note that when routes are exchanged, some customers will necessarily be
+    # removed from the solution because the routes are overlapping. The
+    # offspring are thus incomplete.y
     sol1 = Solution(data, [[3, 4], [1, 2]])
     sol2 = Solution(data, [[2, 3], [4, 1]])
 
     # The start indices do not change because there are no improving moves.
     # So, sol1's route [3, 4] will be replaced by sol2's route [2, 3].
     # This results in two incomplete offspring [[2, 3], [1]] and [[3], [1, 2]],
-    # which are both repaired using greedy repair. After repair, we obtain the
-    # offspring [[2, 3, 4], [1]] with cost 8188, and [[3, 4], [1, 2]] with
-    # cost 9725. The first one is returned since it has the lowest cost.
+    # the first with penalised cost 7898, the second with cost 9495. Thus, the
+    # first one is returned.
     offspring = cpp_srex((sol1, sol2), data, cost_evaluator, (0, 0), 1)
-    expected = Solution(data, [[2, 3, 4], [1]])
+    expected = Solution(data, [[2, 3], [1]])
     assert_equal(offspring, expected)
```

Comment:
This test still has to be updated

Reply:
Since we no longer repair in SREX, we could try to remove OkSmallGreedyRepair.txt too.